samples/simple-chat/csharp/FunctionApp/Functions.cs

```diff
@@ -15,41 +16,58 @@ namespace FunctionApp
+            [SignalRConnectionInfo(HubName = "simplechat", UserId = "{headers.x-ms-signalr-userid}")] SignalRConnectionInfo connectionInfo)
         {
-            // Azure function doesn't support CORS well, workaround it by explicitly return CORS headers
-            req.HttpContext.Response.Headers.Add("Access-Control-Allow-Credentials", "true");
-            if (req.Headers["Origin"].Count > 0) req.HttpContext.Response.Headers.Add("Access-Control-Allow-Origin", req.Headers["Origin"][0]);
-            if (req.Headers["Access-Control-Request-Headers"].Count > 0) req.HttpContext.Response.Headers.Add("Access-Control-Allow-Headers", req.Headers["access-control-request-headers"][0]);
-
-            return new OkObjectResult(connectionInfo);
+            return connectionInfo;
         }
 
         [FunctionName("messages")]
-        public static async Task<IActionResult> SendMessage(
-            [HttpTrigger(AuthorizationLevel.Anonymous, "post", "options")]HttpRequest req,
+        public static Task SendMessage(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "post")]HttpRequest req,
             [SignalR(HubName = "simplechat")]IAsyncCollector<SignalRMessage> signalRMessages)
         {
-            // Azure function doesn't support CORS well, workaround it by explicitly return CORS headers
-            req.HttpContext.Response.Headers.Add("Access-Control-Allow-Credentials", "true");
-            if (req.Headers["Origin"].Count > 0) req.HttpContext.Response.Headers.Add("Access-Control-Allow-Origin", req.Headers["Origin"][0]);
-            if (req.Headers["Access-Control-Request-Headers"].Count > 0) req.HttpContext.Response.Headers.Add("Access-Control-Allow-Headers", req.Headers["access-control-request-headers"][0]);
-
-            if (req.Method == "POST")
-            {
-                var message = new JsonSerializer().Deserialize(new JsonTextReader(new StreamReader(req.Body)));
-
-                await signalRMessages.AddAsync(
-                    new SignalRMessage
-                    {
-                        Target = "newMessage",
-                        Arguments = new[] { message }
-                    });
-            }
-
-            return new OkResult();
+
+            var message = new JsonSerializer().Deserialize<ChatMessage>(new JsonTextReader(new StreamReader(req.Body)));
+
+            return signalRMessages.AddAsync(
+                new SignalRMessage
+                {
+                    UserId = message.recipient,
+                    GroupName = message.groupname,
+                    Target = "newMessage",
+                    Arguments = new[] { message }
+                });
+        }
+
+        [FunctionName("group")]
+        public static Task Group(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "post")]HttpRequest req,
+            [SignalR(HubName = "simplechat")]IAsyncCollector<SignalRGroupAction> signalRGroupActions)
+        {
```

Comment:
It may be more straightforward to provide apis for add and remove from group instead of one that takes a group action parameter.

Reply:
Make sense. Will separate to different API.

---
In reply to: [236927488](https://github.com/Azure/azure-functions-signalrservice-extension/pull/39#discussion_r236927488) [](ancestors = 236927488)